File: xamarin/realm-tutorial-dotnet/Models/Task.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson;
using Realms;

namespace RealmTemplateApp.Models
{
    public class Task : RealmObject
    {
        [PrimaryKey]
        [MapTo("_id")]
        [Required]
        public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

        [MapTo("summary")]
        [Required]
        public string Summary { get; set; }

        [MapTo("isComplete")]
        public bool IsComplete { get; set; }
    }
}

File: sync-todo/v0/client/android-kotlin/app/src/main/java/com/mongodb/app/Task.kt
package com.mongodb.app

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmField
import org.bson.types.ObjectId

open class Task(
        @RealmField("_id") @PrimaryKey var id: ObjectId = ObjectId(),
        var isComplete: Boolean = false,
        var summary: String = ""
): RealmObject() {}
